custom_components/unifiprotect/config_flow.py

```diff
@@ -40,51 +41,93 @@ def async_get_options_flow(config_entry):
         """Get the options flow for this handler."""
         return OptionsFlowHandler(config_entry)
 
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        return await self._init_config_entry(user_input=user_input, reauth=True)
+
     async def async_step_user(self, user_input=None):
         """Handle a flow initiated by the user."""
-        if user_input is None:
-            return await self._show_setup_form(user_input)
+        return await self._init_config_entry(user_input=user_input)
 
-        errors = {}
+    def _get_form(self, reauth: bool):
+        form = self._show_setup_form
+        if reauth:
+            form = self._show_reauth_form
+
+        return form
```

Comment:
```suggestion
        if reauth:
            return self._show_reauth_form
        return self._show_setup_form
```

Reply:
Returning a reference to a function in a config flow isn't a common pattern so this may come back up again in a core review. If we could find a way to avoid it, it would be helpful.